### PR TITLE
Adding Troubleshooting Section

### PR DESCRIPTION
**Pull request by divyenpatel**
Fixes: https://github.com/vmware/kubernetes/issues/206

Customer want us to document the troubleshooting steps with examples of commands. This PR adds details on how to increase log levels, where to find log files to help customer file issue with detailed logs.

Please review: @BaluDontu @pshahzeb @tusharnt

### CODE REVIEW

File: documentation/existing.md [hunk 1]
```diff
@@ -4,6 +4,9 @@ title: Configurations on Existing Kubernetes Cluster
 
 If a Kubernetes cluster has not been deployed using Kubernetes-Anywhere, follow the instructions below to enable the vSphere Cloud Provider. These steps are not needed when using Kubernetes-Anywhere, they will be done as part of the deployment.
 
+We have automated all of the steps mentioned below. For more details, please visit [https://github.com/vmware/kubernetes/blob/enable-vcp-uxi/README.md](https://github.com/vmware/kubernetes/blob/enable-vcp-uxi/README.md). If pre-requisites are not applicable on your setup, then follow below steps manually.
```
Comment (tusharnt) [inline documentation/existing.md]: Technical documentation should not use "we" etc.
Rewording suggestion : "All the steps mentioned below have been documented at "

File: documentation/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
+---
+title: Troubleshooting Kubernetes vSphere Cloud Provider
+---
+This document is aimed to help debug issues in the Kuberentes vSphere Cloud Provider.
```
Comment (tusharnt) [inline documentation/troubleshooting.md]: Why not remove this line? Isn't it redundant?

File: documentation/troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+For general Kubernetes Issue, Please visit [https://kubernetes.io/docs/tasks/debug-application-cluster/debug-cluster/](https://kubernetes.io/docs/tasks/debug-application-cluster/debug-cluster/)
```
Comment (tusharnt) [inline documentation/troubleshooting.md]: For general Kubernetes Issue, Please -> To troubleshoot issues with Kuberenetes cluster deployment, please

File: documentation/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+## What logs can I inspect?
+For digging deeper and help gather data for the technical support we need to see the log files.
```
Comment (tusharnt) [inline documentation/troubleshooting.md]: Please remove this statement as it's redundant.

**Review by shahzebnp — COMMENTED** [review]
Looks good.
What is more helpful is the usage of `kubectl describe` command

As part of troubleshooting, the output of this command on targeted resource like pod, pvc, pv, storageclass can be captured. It helps to narrow down the problem quickly.

eg: 
``` 
kubectl describe pod <podname> --namespace=<namespace_name>
```

File: documentation/troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  journalctl -u kube-controller-manager.service > controller-manager.log
+  ```
+
+Kubelet generally runs as the service and not in the container or Pod in the Kubernetes cluster. Kubelet logs can be obtained using `journalctl` as shown below.
+
+```
+journalctl -u kubelet > kubelet.log
+```
+
+## How can I modify the log level?
```
Comment (tusharnt) [inline documentation/troubleshooting.md]: Let's not use "I"/ "We" in technical documentation.

Rewording suggestion : Modifying log level

File: documentation/troubleshooting.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    .
+    .
+    .
+    - --kubeconfig=/etc/kubernetes/controller-manager.conf
+    - --v=9
+```
+After increasing the log level, you need to restart the `kubelet`, which will restart the API server and Controller-Manager with updated pod manifest file.
+
+
+## What information is needed to provide to technical support for analysis?
```
Comment (tusharnt) [inline documentation/troubleshooting.md]: I am not sure why do we need this section given that logs are collected (mostly) for troubleshooting technical issues.

File: documentation/troubleshooting.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+title: Troubleshooting Kubernetes vSphere Cloud Provider
+---
+This document is aimed to help debug issues in the Kuberentes vSphere Cloud Provider.
+
+For general Kubernetes Issue, Please visit [https://kubernetes.io/docs/tasks/debug-application-cluster/debug-cluster/](https://kubernetes.io/docs/tasks/debug-application-cluster/debug-cluster/)
+
+## What logs can I inspect?
+For digging deeper and help gather data for the technical support we need to see the log files.
+
+For vSphere Cloud Provider most often we need to look into Controller-Manager, Kubelet and API-server logs.
```
Comment (tusharnt) [inline documentation/troubleshooting.md]: We can format this info better. Let's chat offline.

**Review by tusharnt — CHANGES_REQUESTED** [review]

File: documentation/troubleshooting.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+  - command:
+    - kube-controller-manager
+    - --service-account-private-key-file=/etc/kubernetes/pki/sa.key
+    .
+    .
+    .
+    - --kubeconfig=/etc/kubernetes/controller-manager.conf
+    - --v=9
+```
+After increasing the log level, you need to restart the `kubelet`, which will restart the API server and Controller-Manager with updated pod manifest file.
```
Comment (shaominchen) [inline documentation/troubleshooting.md]: Nit: Even if it's straightforward, it's still better to document the command explicitly: /etc/init.d/kubelet restart

Comment (divyenpatel) [reply documentation/troubleshooting.md]: Fixed. Provided command to restart kubelet.
```
systemctl restart kubelet
```

File: documentation/troubleshooting.md (hunk 6 above)
Comment (divyenpatel) [reply documentation/troubleshooting.md]: Removed this section.

File: documentation/troubleshooting.md (hunk 5 above)
Comment (divyenpatel) [reply documentation/troubleshooting.md]: Changed to `Modify log level`

File: documentation/troubleshooting.md (hunk 7 above)
Comment (divyenpatel) [reply documentation/troubleshooting.md]: As discussed provided sections and added proper indentation.

File: documentation/troubleshooting.md (hunk 4 above)
Comment (divyenpatel) [reply documentation/troubleshooting.md]: removed.

File: documentation/existing.md (hunk 1 above)
Comment (divyenpatel) [reply documentation/existing.md]: removed all occurrences of  'i', 'we'.

File: documentation/troubleshooting.md (hunk 2 above)
Comment (divyenpatel) [reply documentation/troubleshooting.md]: removed.

File: documentation/troubleshooting.md (hunk 3 above)
Comment (divyenpatel) [reply documentation/troubleshooting.md]: fixed.

**Comment by divyenpatel** [comment]
@tusharnt Please take a look again. I have addressed all review comments.

File: documentation/troubleshooting.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
+---
+title: Troubleshooting Kubernetes vSphere Cloud Provider
+---
+To troubleshoot issues with Kuberenetes cluster deployment, please visit [https://kubernetes.io/docs/tasks/debug-application-cluster/debug-cluster/](https://kubernetes.io/docs/tasks/debug-application-cluster/debug-cluster/)
+
+## Logs to inspect
+To debug vSphere Cloud Provider Issues, Controller-Manager, API-server and Kubelet logs should be checked.
```
Comment (tusharnt) [inline documentation/troubleshooting.md]: This should be a bullet list as follows :
Following logs are needed to debug VCP issues -
- Controller Manager logs
- API-server logs
- kubelet logs

File: documentation/troubleshooting.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+
+Logs for Controller-Manager and API server should be available on the Master node VM and Logs for Kubelet is located on both Master and Worker Nodes.
```
Comment (tusharnt) [inline documentation/troubleshooting.md]: Let us move this to relevant sections i.e. API server logs and Controller manager logs.

**Review by tusharnt — CHANGES_REQUESTED** [review]

File: documentation/troubleshooting.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
+---
+title: Troubleshooting
+---
+To troubleshoot issues with Kuberenetes cluster deployment, please visit [https://kubernetes.io/docs/tasks/debug-application-cluster/debug-cluster/](https://kubernetes.io/docs/tasks/debug-application-cluster/debug-cluster/)
+
+## Logs to inspect
```
Comment (tusharnt) [inline documentation/troubleshooting.md:6]: Section header can be renamed to "Required logs"

**Review by tusharnt — APPROVED** [review]
LGTM

**Review by tusharnt — APPROVED** [review]

**Pull request merged**